mlrun/model_monitoring/helpers.py

```diff
@@ -447,6 +447,17 @@ def get_result_instance_fqn(
     return f"{model_endpoint_id}.{app_name}.result.{result_name}"
 
 
+def get_name_from_result_fqn(result_fqn: str):
+    if result_fqn.count(".") != 3:
+        raise mlrun.errors.MLRunValueError(
+            f"result_fqn: {result_fqn} is not in the correct format"
+        )
+    # TODO remove after prohibiting "_" in ML-9227
+    result_fqn = result_fqn.replace("_", "-")
+    # Name format cannot contain "."
+    return "--".join(result_fqn.split(".")[i] for i in [0, 1, 3])
```

Comment:
Please add comments here explaining the logic behind this. It's pretty cryptic to see `[0, 1, 3]` without explanation. In general, might want to add a docstring to this function explaining what format of `fqn` is expected and what it does and why.